src/ansys/mechanical/core/embedding/ui.py

```diff
@@ -0,0 +1,185 @@
+        """Open the original mechdb file from save_original().
+
+        Parameters
+        ----------
+        app: ansys.mechanical.core.embedding.app.App
+            A Mechanical embedding application.
+        mechdb_file: pathlib.Path
+            The full path to the active mechdb file.
+        """
+        app.open(mechdb_file)
+
+    def graphically_launch_temp(
+        self, app: "ansys.mechanical.core.embedding.App", mechdb_file: Path, temp_file: Path
+    ) -> Popen:
+        """Launch the GUI for the mechdb file with a temporary name from save_temp_copy().
+
+        Parameters
+        ----------
+        app: ansys.mechanical.core.embedding.app.App
+            A Mechanical embedding application.
+        mechdb_file: pathlib.Path
+            The full path to the active mechdb file.
+        temp_file: pathlib.Path
+            The full path to the temporary mechdb file.
+
+        Returns
+        -------
+        subprocess.Popen
+            The subprocess that launches the GUI for the temporary mechdb file.
+        """
+        p = Popen(
+            [
+                "ansys-mechanical",
+                "--project-file",
+                temp_file,
+                "--graphical",
+                "--revision",
+                str(app.version),
+            ],
+        )
+
+        return p
+
+    def _cleanup_gui(self, process, temp_mechdb_path):
+        cleanup_script = os.path.join(os.path.dirname(os.path.abspath(__file__)), "cleanup_gui.py")
+        Popen([sys.executable, cleanup_script, str(process.pid), temp_mechdb_path])
+
+
+def _is_saved(app: "ansys.mechanical.core.embedding.App") -> bool:
+    """Check if the mechdb file has been saved and raise an exception if not.
+
+    Parameters
+    ----------
+    app: ansys.mechanical.core.embedding.app.App
+        A Mechanical embedding application.
+    """
+    try:
+        app.save()
+    except:
+        raise Exception("The App must have already been saved before using launch_ui!")
+    return True
+
+
+def _launch_ui(
+    app: "ansys.mechanical.core.embedding.App", delete_tmp_on_close: bool, launcher: UILauncher
+) -> None:
+    """Launch the Mechanical UI if the mechdb file has been saved.
+
+    Parameters
+    ----------
+    app: ansys.mechanical.core.embedding.app.App
+        A Mechanical embedding application.
+    delete_tmp_on_close: bool
+        Whether to delete the temporary mechdb file when the GUI is closed.
+        By default, this is ``True``.
+    launcher: UILauncher
+        Launch the GUI using a temporary mechdb file.
+    """
+    if _is_saved(app):
+        launcher.save_original(app)
+        mechdb_file, temp_file = launcher.save_temp_copy(app)
+        launcher.open_original(app, mechdb_file)
+        p = launcher.graphically_launch_temp(app, mechdb_file, temp_file)
+
+        # If the user wants the temporary file to be deleted
+        if delete_tmp_on_close:
+            atexit.register(launcher._cleanup_gui, p, temp_file)
```

Comment:
this doesn't need to be onexit, I think _cleanup_gui can be run immediately

Reply:
If it's run immediately, wouldn't it affect the launched gui if it's still open and we're deleting the file it's using?